tube/etl/indexers/aggregation/translator.py

```diff
@@ -125,19 +125,39 @@ def get_direct_children(self, root_df):
             child_by_root.unpersist()
         return root_df
 
-    def get_joining_props(self, translator, joining_index):
+    def get_joining_props(self, translator, joining_index, with_agg_func):
         """
         Get joining props added by an additional join between indices/documents
         :param joining_index: Joining index created from parser
         :return:
         """
         props = []
         for r in joining_index.getting_fields:
+            if not (with_agg_func ^ (r.fn is None)):
```

Comment:
why need bit operator here? what's the difference from using just `or`?

Reply:
it is equivalent to `(with_agg_func and r.fn is None) or (not with_agg_func and r.fn is not None)`. The way of writing `and` and `or` is more complex

Reply:
it's so hard to read though, can we just make this return `{props_with_fn: [], props_without_fn: []}` and call it once?

Reply:
yeah, good suggestion. I updated